backend/src/schema/resolvers/users.spec.js

```diff
@@ -273,141 +273,284 @@ describe('DeleteUser', () => {
       }
     }
   `
-  beforeEach(async () => {
-    variables = { id: ' u343', resource: [] }
+  describe('as another user', () => {
+    beforeEach(async () => {
+      variables = { id: ' u343', resource: [] }
 
-    user = await Factory.build('user', {
-      name: 'My name should be deleted',
-      about: 'along with my about',
-      id: 'u343',
+      user = await Factory.build('user', {
+        name: 'My name should be deleted',
+        about: 'along with my about',
+        id: 'u343',
+      })
+    })
+
+    beforeEach(async () => {
+      const anotherUser = await Factory.build(
+        'user',
+        {
+          role: 'user',
+        },
+        {
+          email: 'user@example.org',
+          password: '1234',
+        },
+      )
+
+      authenticatedUser = await anotherUser.toJson()
     })
-    await Factory.build(
-      'user',
-      {
-        id: 'not-my-account',
-      },
-      {
-        email: 'friends-account@example.org',
-      },
-    )
-  })
 
-  describe('unauthenticated', () => {
-    it('throws authorization error', async () => {
+    it("an ordinary user has no authorization to delete another user's account", async () => {
       const { errors } = await mutate({ mutation: deleteUserMutation, variables })
       expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
     })
   })
 
-  describe('authenticated', () => {
+  describe('as moderator', () => {
     beforeEach(async () => {
-      authenticatedUser = await user.toJson()
-    })
+      variables = { id: ' u343', resource: [] }
 
-    describe("attempting to delete another user's account", () => {
-      beforeEach(() => {
-        variables = { ...variables, id: 'not-my-account' }
+      user = await Factory.build('user', {
+        name: 'My name should be deleted',
+        about: 'along with my about',
+        id: 'u343',
       })
+    })
+
+    beforeEach(async () => {
+      const moderator = await Factory.build(
+        'user',
+        {
+          role: 'moderator',
+        },
+        {
+          email: 'moderator@example.org',
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-13T13:48:59Z" title="Thursday, February 13th 2020, 2:48:59 pm +01:00">Feb 13, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -568,3 +568,330 @@ describe('DeleteUser', () => {
     })
   })
 })
+
+
+
+describe('DeleteUser as Admin', () => {
+  const deleteUserMutation = gql`
+    mutation($id: ID!, $resource: [Deletable]) {
+      DeleteUser(id: $id, resource: $resource) {
+        id
+        name
+        about
+        deleted
+        contributions {
+          id
+          content
+          contentExcerpt
+          deleted
+          comments {
+            id
+            content
+            contentExcerpt
+            deleted
+          }
+        }
+        comments {
+          id
+          content
+          contentExcerpt
+          deleted
+        }
+      }
+    }
+  `
+  beforeEach(async () => {
+    variables = { id: ' u343', resource: [] }
+
+    user = await Factory.build('user', {
+      name: 'My name should be deleted',
+      about: 'along with my about',
+      id: 'u343',
+    })
+    await Factory.build(
+      'user',
+      {
+        id: 'not-my-account',
+      },
+      {
+        email: 'friends-account@example.org',
+      },
+    )
+  })
+
+  describe('unauthenticated', () => {
+    it('throws authorization error', async () => {
+      const { errors } = await mutate({ mutation: deleteUserMutation, variables })
+      expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
+    })
+  })
+
+  describe('authenticated as Admin', () => {
+    beforeEach(async () => {
+      const admin = await Factory.build(
+        'user',
+        {
+          role: 'admin',
+        },
+        {
+          email: 'admin@example.org',
+          password: '1234',
+        },
+      )
+      authenticatedUser = await admin.toJson()
+    })
+ 
+
+    describe('attempting to delete a foreign account by an Admin', () => {
```


```suggestion
    describe('deleting a user account', () => {
```
this would then read like `authenticated as Admin deleting a user account`